### PR TITLE
 Perf: Traverse commit history instead of reading the packfile

### PR DESCRIPTION
**Pull request by ajnavarro**
To avoid show orphan commits and improve performance on commits table, instead of reading all the objects from each repository and discard commits, we can execute from go-git a git log --all command and traverse commit history, avoiding read a lot of data from disk.

It closes https://github.com/src-d/gitbase/issues/617

### CODE REVIEW

**Comment by kuba--** [comment]
Let's fix tests, first.

File: commits.go [hunk 1]
```diff
@@ -161,7 +162,10 @@ func (i *commitIter) init() error {
 	if len(i.hashes) > 0 {
 		i.iter, err = NewCommitsByHashIter(i.repo, i.hashes)
 	} else {
-		i.iter, err = i.repo.CommitObjects()
+		i.iter, err =
+			i.repo.Log(&git.LogOptions{
+				All: true,
```
Comment (kuba--) [inline commits.go]: Maybe we should force order by time (WDYT)?
By default  pre-order will be used, which I think works slightly different than native git implementation.

**Comment by ajnavarro** [comment]
@kuba-- @erizocosmico tests finally passing

File: commits.go (hunk 1 above)
Comment (ajnavarro) [reply commits.go]: but here we don't want to mimic 100% git log output, we want to reduce unnecessary disk reads. Ordering commits will not give us any performance improvement.

**Comment by ajnavarro** [comment]
Some numbers using gitbase-regression with complexity=2:
```
ID      | remote:master   | local:perf/list-only-reachable-objects 
 query00 | 4.012231112s    | 1.482114222s 
 query01 | 115.760157ms    | 114.007921ms 
 query02 | 1.022439327s    | 1.008603787s 
 query03 | 937.115316ms    | 931.223158ms 
 query04 | 3.701316738s    | 1.361501136s 
 query05 | 1.377949809s    | 1.298451385s 
 query06 | 11m14.10874939s | 10m27.993923365s 
 query07 | 4.14140327s     | 1.388045549s 
 query08 | 4.193708919s    | 1.375377416s 
 query09 | 4.168560944s    | 1.395838104s 
 query10 | 6.991478009s    | 3.85651492s 
 query11 | 10m12.68998881s | 10m6.052184916s 
 query12 | 4.961075222s    | 4.957561797s 
 query13 | 116.180972ms    | 110.658067ms 
```

**Pull request merged**